lib/srv/sessiontracker.go

```diff
@@ -181,6 +182,33 @@ func (s *SessionTracker) WaitForStateUpdate(initialState types.SessionState) typ
 	}
 }
 
+// SubscribeToStateUpdate waits for the tracker's state to be updated and sends the new states over a channel
+func (s *SessionTracker) SubscribeToStateUpdate(ctx context.Context) <-chan types.SessionState {
```

Comment:
One API that might be cleaner would be

```
func (s *SessionTracker) WaitForState(ctx context.Context, state types.SessionState) ```

With this approach, you don't need the `math.MaxInt32` in there, and the caller doesn't have to deal with channels.

Reply:
I considered this but it doesn't work. It would be similar to the other function we already have. The issue is we need a channel to be able to select in the top level test to handle a timeout. With an API like this, a timeout isn't doable.

Reply:
@zmb3 I was able to construct a slightly better API, it's a bit awkward in the way that you need to ping the condvar for the timeout to take effect but we're now able to get rid of the channel.